src/middlewared/middlewared/plugins/directoryservices.py

```diff
@@ -344,6 +368,34 @@ def get_last_password_change(self, domain=None):
 
         return {"dbconfig": stored_ts, "secrets": passwd_ts}
 
+    @private
+    def get_conf(self, schema, ha_mode=None):
+        if ha_mode is None:
+            ha_mode = self.middleware.call_sync('smb.get_smb_ha_mode')
+        with DirectorySecrets(logger=self.logger, ha_mode=ha_mode) as s:
+            conf = s.get_conf(schema)
+
+        rv = conf.decode()
```

Comment:
This will raise an `AttributeError` if rv is `None`.

```suggestion
        rv = conf.decode() if conf is not None else {}
```